pkg/tcpip/transport/tcp/endpoint.go

```diff
@@ -3208,7 +3243,7 @@ func (e *endpoint) Stats() tcpip.EndpointStats {
 
 // Wait implements stack.TransportEndpoint.Wait.
 func (e *endpoint) Wait() {
-	waitEntry, notifyCh := waiter.NewChannelEntry(waiter.EventHUp)
+	waitEntry, notifyCh := waiter.NewChannelEntry(waiter.EventHUp | waiter.EventRdHUp)
```

Comment:
I believe your PR needs to be rebased @ HEAD. This code has changed. Further I don't think we care about RdHup here as we do want the endpoint to be fully closed when Wait returns.

Reply:
Removed and rebased.